pkg/virtual/workspaces/builder/org_listener.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/authentication/user"
+
+	"github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1/helper"
+	workspaceauth "github.com/kcp-dev/kcp/pkg/virtual/workspaces/auth"
+	workspacecache "github.com/kcp-dev/kcp/pkg/virtual/workspaces/cache"
+	virtualworkspacesregistry "github.com/kcp-dev/kcp/pkg/virtual/workspaces/registry"
+)
+
+// orgListener listens for changes in the root WorkspaceAuthCache,
+// in order to manage the list of orgs
+type orgListener struct {
+	newOrg func(orgName string) virtualworkspacesregistry.Org
+
+	clusterWorkspaceCache *workspacecache.ClusterWorkspaceCache
+
+	knownWorkspaces   sets.String
```

Comment:
is this protected? Does it have to be?

Reply:
what does it contain? What does "known" mean?

Reply:
Answered and fixed in commit 9c3f0facee5da12f6440fb1bccee131bfc0180d4